packages/lab/src/stepper-input/useStepperInput.ts

```diff
@@ -54,9 +62,20 @@ export const useStepperInput = (
 
   const [currentValue, setCurrentValue, isControlled] = useControlled({
     controlled: value,
-    default: toFixedDecimalPlaces(defaultValue, decimalPlaces),
+    default: defaultValue,
     name: "stepper-input",
   });
+
+  const [inputValue, setInputValue] = useState<number | string>(
+    toFixedDecimalPlaces(defaultValue, decimalPlaces)
+  );
+
+  useEffect(() => {
```

Comment:
Why is this needed? Will this cause infinite loop? Edge case for `useControlled` ?

Reply:
because the input value is separate from the stepper input value so it needs to stay up to date. I tried to use useControlled first

Reply:
state update should be in sync with events that triggered them, not via `useEffect` in another render cycle

Reply:
you're right, I think I've managed to remove it and can't find any issues

Reply:
oh actually the same issue is there, if the value is controlled it doesn't update without this. See this example: https://628e27029fcf95004a62cafd-sipdhyjpaz.chromatic.com/?path=/story/lab-stepper-input--refresh-adornment